### PR TITLE
Documentation for `torch.optim.swa_utils`

### PR DESCRIPTION
**Pull request by izmailovpavel**
This PR adds a description of `torch.optim.swa_utils` added in https://github.com/pytorch/pytorch/pull/35032 to the docs at `docs/source/optim.rst`. Please let me know what you think!

@vincentqb @andrewgordonwilson

### CODE REVIEW

**Review by vincentqb — APPROVED** [review]
LGTM!

Let's see what this looks like [here](https://pytorch.org/docs/master/optim.html), once the PR has landed.

**Comment by vincentqb** [comment]
cc @jlin27 to add to the release

File: docs/source/optim.rst [hunk 1]
```diff
@@ -175,3 +175,107 @@ should write your code this way:
...
+>>> swa_start = 160
+>>> swa_scheduler = SWALR(optimizer, swa_lr=0.05)
+>>>
+>>> for epoch in range(300):
+>>>       for input, target in loader:
+>>>           optimizer.zero_grad()
+>>>           loss_fn(model(input), target).backward()
+>>>           optimizer.step()
+>>>       if i > swa_start:
+>>>           swa_model.update_parameters(model)
```
Comment (Daniil-Osokin) [inline docs/source/optim.rst:273]: @izmailovpavel Hi, thanks for the implementation of SWA in PyTorch! Do you find it useful to average the weights during annealing phase? By default `anneal_epochs` equals to 10 (and cannot be set to 0), so in this snippet for the first 10 epochs after `swa_start` weights will be averaged during annealing phase.

Comment (izmailovpavel) [reply docs/source/optim.rst:273]: Hey Daniil, thank you for looking into the implementation! Generally, it's reasonable to average the weights during the annealing phase, but I imagine there could be cases when it's not desirable, e.g. when the learning rate before the annealing is way too high

Comment (Daniil-Osokin) [reply docs/source/optim.rst:273]: Thanks for the response! It is just confusing a bit. According to the paper, the purpose of SWA is to average weights during exploration of loss surface (to find high-performing networks), but averaging during annealing would lead to averaging with different (decreasing) learning rates, which paper refers as non suitable (links to experiments in Ruppert, 1988) for improving generalization, due to SGD does not perform very differently under this schedule.

Comment (izmailovpavel) [reply docs/source/optim.rst:273]: It may be safer to fix it in the docs as you suggest.

```
lr_anneal_epochs = 10
swa_scheduler = SWALR(optimizer, swa_lr=0.05, anneal_epochs=lr_anneal_epochs)
...

if i > swa_start + lr_anneal_epochs:
    swa_model.update_parameters(model)
```
It makes the example a bit more complex... @vincentqb @andrewgordonwilson what do you think?

Comment (Daniil-Osokin) [reply docs/source/optim.rst:273]: Is any chance to remove annealing phase from the scheduler? I believe this is the part which accidentally differs from the paper.

Comment (vincentqb) [reply docs/source/optim.rst:273]: > It may be safer to fix it in the docs as you suggest.
> 
> ```
> lr_anneal_epochs = 10
> swa_scheduler = SWALR(optimizer, swa_lr=0.05, anneal_epochs=lr_anneal_epochs)
> ...
> 
> if i > swa_start + lr_anneal_epochs:
>     swa_model.update_parameters(model)
> ```
> 
> It makes the example a bit more complex... @vincentqb @andrewgordonwilson what do you think?

What would be an alternate fix that would make this simpler?

Comment (Daniil-Osokin) [reply docs/source/optim.rst:273]: I believe the simplest option is to allow SWALR take `anneal_epochs=0` by default, thus skip annealing phase. In this case this snippet will be consistent to the paper and can be left as is (but it requires SWALR code modification). Can make a PR if it will work.

Comment (izmailovpavel) [reply docs/source/optim.rst:273]: Sorry for not following up on this discussion for so long. @Daniil-Osokin I agree that we should allow `anneal_epochs=0`, and I think it would be great if you could make a pull request for that. I am not sure if we should by default set the number of annealing epochs to zero or not, I am up for discussion. I think both options are reasonable, but I agree that it may be better to set `anneal_epochs=0` by default.
